frontend/src/hooks/useWidgetPaging.js

```diff
@@ -0,0 +1,180 @@
+import { useEffect, useState } from 'react';
+import { DECIMAL_BASE } from '@ttahub/common';
+import useSessionSort from './useSessionSort';
+
+export const parseValue = (value) => {
+  const noCommasValue = value.replaceAll(',', '');
+  const parsedValue = parseInt(noCommasValue, DECIMAL_BASE);
+  if (Number.isNaN(parsedValue)) {
+    return value;
+  }
+  return parsedValue;
+};
+
+export default function useWidgetPaging(
+  headers,
+  localStorageKey,
+  defaultSortConfig,
+  perPageNumber,
+  dataToUse,
+  setDataToUse,
+  resetPagination,
+  setResetPagination,
+  loading,
+  checkBoxes,
+  exportHeading,
+  setDataPerPage,
+  stringColumns = [],
+  dateColumns = [],
+) {
+  const [sortConfig, setSortConfig] = useSessionSort(defaultSortConfig, localStorageKey);
+  const { activePage } = sortConfig;
+  const [offset, setOffset] = useState((activePage - 1) * perPageNumber);
+
+  // a side effect that resets the pagination when the filters change
+  useEffect(() => {
+    if (resetPagination) {
+      setSortConfig({ ...sortConfig, activePage: 1 });
+      setOffset(0); // 0 times perpage = 0
+      setResetPagination(false);
+    }
+  }, [activePage, resetPagination, setResetPagination, setSortConfig, sortConfig]);
+
+  const handlePageChange = (pageNumber) => {
+    if (!loading) {
+      // copy state
+      const sort = { ...sortConfig };
+
+      // mutate
+      sort.activePage = pageNumber;
+
+      // store it
+      setSortConfig(sort);
+      setOffset((pageNumber - 1) * perPageNumber);
+    }
+  };
+
+  const requestSort = (sortBy) => {
+    // Get sort direction.
+    let direction = 'asc';
+    if (
+      sortConfig
+      && sortConfig.sortBy === sortBy
+      && sortConfig.direction === 'asc'
+    ) {
+      direction = 'desc';
+    }
+
+    const sortingByStringColumn = stringColumns.includes(sortBy);
+    const sortingByDateColumn = dateColumns.includes(sortBy);
```

Comment:
worked when I tried it, anyways

Reply:
If you think about it, you can sort ```"09/01/2023" & "09/02/2023"``` or ```"2022-01-02" & "2023-01-02"``` as strings and that would come out correctly. The only time it wouldn't work is if the dates are in a different format, which is a problem to solve elsewhere IMO

Reply:
I think the issue occurred when comparing something like 01/01/2024 to 12/01/2023. It thought that 2023 was before 2024 because it was a higher month.

Reply:
oh yeah you are right